controllers/managedClusterResources.go

```diff
@@ -0,0 +1,468 @@
+
+	config := ctrl.GetConfigOrDie()
+	dynamic := dynamic.NewForConfigOrDie(config)
+
+	for _, item := range templates {
+		r.Log.Info("[createResourcesFromTemplates]", "cluster", data.Cluster, "template", item.resourceName)
+		obj := &unstructured.Unstructured{}
+		w, err := r.renderYamlTemplate(item.resourceName, item.template, *data)
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return err
+		}
+
+		// decode YAML into unstructured.Unstructured
+		dec := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+		_, gvk, err := dec.Decode(w.Bytes(), nil, obj)
+		if err != nil {
+			return err
+		}
+		// Map GVK to GVR with discovery client
+		discoveryClient, err := discovery.NewDiscoveryClientForConfig(config)
+		if err != nil {
+			return err
+		}
+		mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
+		mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+		if err != nil {
+			return err
+		}
+		// Build resource
+		resource := schema.GroupVersionResource{
+			Group:    gvk.Group,
+			Version:  gvk.Version,
+			Resource: mapping.Resource.Resource,
+		}
+		_, err = dynamic.Resource(resource).Namespace(data.Cluster).Create(ctx, obj, metav1.CreateOptions{})
+		if err != nil {
+			if errors.IsAlreadyExists(err) {
+				r.Log.Info("[createResourcesFromTemplates] Already exists",
+					"cluster", data.Cluster, "template", item.resourceName)
+				return nil
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+// deleteManagedClusterViewResource deletes view by name and namespace
+// returns: error
+func (r *ClusterGroupUpgradeReconciler) deleteManagedClusterViewResource(
+	ctx context.Context, name string, namespace string) error {
+
+	config := ctrl.GetConfigOrDie()
+	dynamic := dynamic.NewForConfigOrDie(config)
+	resourceID := schema.GroupVersionResource{
+		Group:    "view.open-cluster-management.io",
+		Version:  "v1beta1",
+		Resource: "managedclusterviews",
+	}
+	err := dynamic.Resource(resourceID).Namespace(namespace).Delete(
+		ctx, name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getView gets view resource
+// returns:     *unstructured.Unstructured (view data)
+//              bool (available)
+//              error
+func (r *ClusterGroupUpgradeReconciler) getView(
+	ctx context.Context, name string, namespace string) (
+	*unstructured.Unstructured, bool, error) {
+
+	view := &unstructured.Unstructured{}
+	view.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "view.open-cluster-management.io",
+		Kind:    "ManagedClusterView",
+		Version: "v1beta1",
+	})
+	err := r.Client.Get(ctx, client.ObjectKey{
+		Name:      name,
+		Namespace: namespace,
+	}, view)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return view, false, nil
+		}
+		return view, false, err
+	}
+	return view, true, nil
+}
+
+// deleteAllViews deletes all managed cluster view resources
```

Comment:
No, it only deletes the view. The namespace is the managed cluster namespace - it will be terrible to delete it. But the view itself is a namespaced resource, maybe that is the confusion:
```go
// deleteManagedClusterViewResource deletes view by name and namespace
// returns: error
func (r *ClusterGroupUpgradeReconciler) deleteManagedClusterViewResource(
	ctx context.Context, name string, namespace string) error {

	config := ctrl.GetConfigOrDie()
	dynamic := dynamic.NewForConfigOrDie(config)
	resourceID := schema.GroupVersionResource{
		Group:    "view.open-cluster-management.io",
		Version:  "v1beta1",
		Resource: "managedclusterviews",
	}
	err := dynamic.Resource(resourceID).Namespace(namespace).Delete(
		ctx, name, metav1.DeleteOptions{})
	if err != nil {
		return err
	}
	return nil
}
```